### ISSUE TITLE
etcd: query timeout due to nonexistent "lease" node

### ISSUE THREAD

**Issue by gedgar** [vitessio/vitess#6258]
#### Overview of the Issue

User is running Vitess (latest) in a docker container locally for testing purposes. Each test creates some tables, loads schema and vschema, and at the end does cleanup by dropping tables and loading an empty vschema. 

After attempting multiple queries in succession, they're encountering issues with the last cleanup query, which executes `DROP TABLE` via `vtctl ApplySchema`

#### Reproduction Steps
1. Load attached schemas and vschemas. 
2. On MySql connection issue the following twice:
```
INSERT INTO `Testing.TimeoutRepros` (`uk`) VALUES (2);
SELECT `id`, `uk` FROM `Testing.TimeoutRepros` WHERE `id`=last_insert_id();
```
3. Immediately (I have a command waiting and ready in another window) issue a command to drop the table via vtctl client ApplySchema
```DROP TABLE `Testing.TimeoutRepros`;"```

4. After approximately 30 seconds You will see a transaction timeout in vitess logs, I actually see 2 of them:
```
W0526 23:52:15.888677    1368 tx_pool.go:127] killing transaction (exceeded timeout: 30s): 1590536965992064103	'principal:"ochotinun" component:"172.17.0.1:55332" subcomponent:"VTGate MySQL Connector" '	'username:"userData1" '	May 26 23:51:44.306985	Jan  1 00:00:00.000000	-9223372036.854776			
W0526 23:52:15.889755    1368 tx_pool.go:127] killing transaction (exceeded timeout: 30s): 1590536965992064104	'principal:"ochotinun" component:"172.17.0.1:55332" subcomponent:"VTGate MySQL Connector" '	'username:"userData1" '	May 26 23:51:44.312102	Jan  1 00:00:00.000000	-9223372036.854776
```

5. My MySql connection now can't issue any commands for a few minutes:
```
mysql> show tables;
ERROR 1317 (HY000): vtgate: http://9f4cadc25827:14999/: target: sharded.-80.master, used tablet: docker-2 (localhost): vttablet: rpc error: code = Aborted desc = transaction 1590536965992064104: ended at 2020-05-26 23:52:15.889 UTC (exceeded timeout: 30s) (CallerID: userData1)
```

From the logs, it looks like the table was dropped successfully but we encountered the familiar `node doesn't exist:` error for `lease`:
```
I0524 21:16:48.561777    2051 engine.go:187] Time taken to load the schema: 4.17ms
I0524 21:16:48.567758      50 locks.go:252] Unlocking keyspace sharded for successful action ApplySchemaKeyspace
I0524 21:16:48.573919      50 plain_controller.go:103] Executor finished, result: {
  "FailedShards": [],
  "SuccessShards": [
    {
      "Shard": "-80",
      "Result": {},
      "Position": "MySQL56/9244fcfe-9e03-11ea-82ec-0242ac110002:1-21"
    },
    {
      "Shard": "80-",
      "Result": {},
      "Position": "MySQL56/9586fa3e-9e03-11ea-b8ec-0242ac110002:1-20"
    }
  ],
  "CurSQLIndex": 0,
  "Sqls": [
    "DROP TABLE `AllTypes.AllTypesEntity`"
  ],
  "ExecutorErr": "node doesn't exist: lease",
  "TotalTimeSpent": 30696896200
}
W0524 21:16:54.476956    1371 tabletserver.go:1513] Code: ABORTED
transaction 1590354880437398403: ended at 2020-05-24 21:16:48.368 UTC (exceeded timeout: 30s)
: Sql: "rollback", BindVars: {}
W0524 21:16:54.476956    2051 tabletserver.go:1513] Code: ABORTED
transaction 1590354886627646603: ended at 2020-05-24 21:16:48.549 UTC (exceeded timeout: 30s)
: Sql: "rollback", BindVars: {}
W0524 21:16:54.476956    2051 tabletserver.go:1513] Code: ABORTED
transaction 1590354886627646602: ended at 2020-05-24 21:16:48.548 UTC (exceeded timeout: 30s)
: Sql: 
```

See attached [vitess_log.txt](https://github.com/vitessio/vitess/files/4718713/vitess_log.txt) for more logs.

**Comment by yborovikov** [comment]
to reproduce:
(1) start `vttablet`s with `-queryserver-config-transaction-timeout 60` option
(2) try to insert a row with `null` for a non-null column (this should fail in mysql)
(3) try to drop the table

(3) will time out (30s by default) because it's waiting for the stale transaction from (2) which takes 60s (as set by `queryserver-config-transaction-timeout` option) to timeout.

as an alternative to (1) you can set an explicit timeout on query (3) instead - say, 15s (default is 30s) - so that it's less than stale transaction timeout (which seems to also be 30s).

the problem likely happens because `vttablet` doesn't seem to properly end the transactions that fail on mysql side.

relevant log lines on vitess side (with `-queryserver-config-transaction-timeout 5` option):
```
I0605 01:00:50.864353    1358 tabletserver.go:1518] Column 'user_id' cannot be null (errno 1048) (sqlstate 23000) (CallerID: userData1): Sql: "insert into `Roblox.Testing.NoNullFields`(id, user_id) values (:_id0, :_user_id0)", BindVars: {__seq0: "type:INT64 value:\"1\" "_id0: "type:INT64 value:\"1\" "_user_id0: ""}
[...]
W0605 01:00:56.137078    2031 tx_pool.go:127] killing transaction (exceeded timeout: 5s): 1591318336990850107	'component:"172.17.0.1:33212" subcomponent:"VTGate MySQL Connector" '	'username:"userData1" '	Jun  5 01:00:50.866356	Jan  1 00:00:00.000000	-9223372036.854776
```

**Comment by gedgar** [comment]
Fixed by #6299
